### DISCUSSION TITLE
Get HTTP play_file to return immediately?

### DISCUSSION THREAD

**Discussion by SteveNewhouse** [avantrec/soco-cli#75] [Q&A]
Hello- I'm calling the SOCO HTTP server from a "Get Contents of URL" action in an iOS shortcut to play a file.  Unfortunately the shortcut waits for the entire song to finish before continuing (which is not what I want) because the HTTP server does not provide a response until then.  Is there a way to arrange it so that the server does not block while the song is playing?

**Comment by pwt** [comment]
This should be possible. I'll have a think about what the right design / user model should be.

**Reply by pwt** [reply]
SoCo-CLI v0.4.77 adds the ability to [run HTTP API actions asynchronously](https://github.com/avantrec/soco-cli?tab=readme-ov-file#asynchronous-commands-experimental).

You simply replace the `play_file` action with `async_play_file`.

I'd consider this experimental for time being. Please let me know if it works for you.

**Reply by SteveNewhouse** [reply]
Wow... thank you.  I will give it a go right now.

**Reply by SteveNewhouse** [reply]
Works great- thank you for updating so quickly that is amazing!  Wonderful thing you've built here.

**Reply by pwt** [reply]
Great! Thanks for testing.
